### DISCUSSION TITLE
useModel有沒有類型redux devtools的管理工具？

### DISCUSSION THREAD

**Discussion by YaoKaiPeng** [umijs/umi#12153] [Q&A]
useModel有沒有類型redux devtools的管理工具？

**Comment by fz6m** [comment]
没有。目前不推荐继续使用 use request 、use model 、dva 等历史数据方案，这些方案是为了旧项目升级而保留下来的，目前已经过气被淘汰了，现代前端数据流更推荐使用 [react-query](https://umijs.org/docs/max/react-query) 或 [swr](https://github.com/vercel/swr) 管理请求状态，使用 [jotai](https://github.com/pmndrs/jotai) 或 [valtio](https://umijs.org/docs/max/valtio) 管理全局状态。
